packages/express-react/src/server/preview.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  type Request as ExpressRequest,
+  type Response as ExpressResponse,
+  type RequestHandler,
+  type NextFunction,
+} from 'express'
+
+import {
+  type SiteVersion,
+  MAKESWIFT_SITE_VERSION_COOKIE,
+  REDIRECT_SEARCH_PARAM,
+  cookieSettingOptions,
+  MakeswiftClient,
+  pipeResponseTo,
+  redirectLiveHandler,
+  SearchParams,
+  secondsUntilSiteVersionExpiration,
+  serializeSiteVersion,
+  toApiRequest,
+} from '@makeswift/runtime/unstable-framework-support'
+
+async function requestedSiteVersion({
+  searchParams,
+  client,
+}: {
+  searchParams: URLSearchParams
+  client: MakeswiftClient
+}): Promise<SiteVersion | null> {
+  const previewToken = searchParams.get(SearchParams.PreviewToken)
+  if (previewToken == null) return null
+
+  const verifiedToken = await client.readPreviewToken(previewToken)
+  if (verifiedToken == null) return null
+
+  const { payload } = verifiedToken
+
+  return { version: payload.version, token: previewToken }
+}
+
+export function createPreviewMiddleware({ client }: { client: MakeswiftClient }): RequestHandler {
+  return async function apiHandler(req: ExpressRequest, res: ExpressResponse, next: NextFunction) {
+    const { pathname, searchParams } = new URL(req.url, `https://example.com`)
+
+    const redirectLive = searchParams.get(REDIRECT_SEARCH_PARAM)
+    if (redirectLive != null) {
+      return pipeResponseTo(
+        await redirectLiveHandler(toApiRequest(req), {
+          previewCookieNames: [MAKESWIFT_SITE_VERSION_COOKIE],
+        }),
+        res,
+      )
+    }
+
+    const siteVersion = await requestedSiteVersion({ searchParams, client })
+    if (siteVersion == null) return next()
+
+    const serializedSiteVersion = serializeSiteVersion(siteVersion)
+    const secondsUntilExpiration = secondsUntilSiteVersionExpiration(siteVersion)
+
+    res.cookie(MAKESWIFT_SITE_VERSION_COOKIE, serializedSiteVersion, {
+      ...cookieSettingOptions,
+      maxAge: secondsUntilExpiration,
```

Comment:
I think `res.cookie` `maxAge` is in [milliseconds](https://expressjs.com/en/api.html)?
<img width="866" height="322" alt="image" src="https://github.com/user-attachments/assets/dbc2cb20-8a69-4ef4-898c-662a22129069" />

Reply:
It's extra funny because if both `Expires` and `Max-Age` are set, `Max-Age` has precedence, so the appearance of the greater precision that this interface gives is well, an illusion.